vetiver/tests/test_write_app.py

```diff
@@ -1,38 +1,39 @@
-import os
 import pins
 import vetiver
+import pytest
+from tempfile import TemporaryDirectory
+from pathlib import Path
 
-# Load data, model
-X_df, y = vetiver.get_mock_data()
-model = vetiver.get_mock_model().fit(X_df, y)
 
+@pytest.fixture
+def vetiver_model_creation():
+    X_df, y = vetiver.get_mock_data()
+    model = vetiver.get_mock_model().fit(X_df, y)
+    return vetiver.VetiverModel(model, "model")
 
-def test_write_app():
-    file = "app.py"
-    v = vetiver.VetiverModel(
-        model=model, prototype_data=X_df, model_name="model", versioned=None
-    )
-    model_board = pins.board_folder(path=".", versioned=True, allow_pickle_read=True)
-    vetiver.vetiver_pin_write(board=model_board, model=v)
-    vetiver.write_app(model_board, "model", file="app.py")
-    contents = open(file).read()
-    os.remove(file)
-    version = model_board.pin_versions("model").sort_values(
-        by="created", ascending=False
-    )
-    version = version.version[0]
-    assert (
-        contents
-        == f"""from vetiver import VetiverModel
+
+def test_write_app(vetiver_model_creation):
+    with TemporaryDirectory() as tempdir:
```

Comment:
This required an update to be in a temporary directory otherwise pytest + pins kept the same "abc" protocol from earlier tests, which cannot be deparsed.

Reply:
@pytest.fixture(scope='module') to the rescue!